sycl/source/detail/queue_impl.cpp

```diff
@@ -310,6 +310,57 @@ void queue_impl::addEvent(const event &Event) {
   }
 }
 
+event queue_impl::submit_impl(const detail::type_erased_cgfo_ty &CGF,
+                              const std::shared_ptr<queue_impl> &Self,
+                              queue_impl *SecondaryQueue, bool CallerNeedsEvent,
+                              const detail::code_location &Loc,
+                              bool IsTopCodeLoc,
+                              const SubmissionInfo &SubmitInfo) {
+  handler Handler(Self, SecondaryQueue, CallerNeedsEvent);
+  auto &HandlerImpl = detail::getSyclObjImpl(Handler);
+  if (xptiTraceEnabled()) {
```

Comment:
I have a fix [here](https://github.com/intel/llvm/pull/18234), but just for awareness `xptiTraceEnabled` isn't defined when `XPTI_ENABLE_INSTRUMENTATION=OFF` so it needs to be ifdefd.